python/fedml/cross_silo/client/fedml_client_master_manager.py

```diff
@@ -103,8 +108,9 @@ def handle_message_receive_model_from_server(self, msg_params):
             self.__train()
             self.round_idx += 1
         else:
-            self.send_client_status(0, ClientMasterManager.RUN_FINISHED_STATUS_FLAG)
-            mlops.log_training_finished_status()
+            if self.is_main_process():
+                self.send_client_status(0, ClientMasterManager.RUN_FINISHED_STATUS_FLAG)
```

Comment:
there are multiple locations we need to call "send_client_status". Please directly add the main process filter in it.

Reply:
Updated in commit https://github.com/FedML-AI/FedML/pull/1091/commits/285be7f0596f5d8ea437ed36bf3df367bccd912e, please take a look